source/fundamentals/object-encodings.txt

```diff
@@ -0,0 +1,98 @@
+.. default-domain:: mongodb
+
+Overview
+--------
+
+You can customize the class that represents BSON documents in a collection
+using an optional second parameter of the
+:java-sync-api:`MongoDatabase.getCollection()
+<com/mongodb/client/MongoDatabase.html#getCollection(java.lang.String,java.lang.Class)>`
+method.
+
+To use a class in this way, a :java-docs:`Codec
+<apidocs/bson/org/bson/codecs/Codec.html>` for it must be registered in the
+:java-docs:`CodecRegistry
+<apidocs/bson/org/bson/codecs/configuration/CodecRegistry.html>`
+of the ``MongoCollection``.
+
+Default Codecs
+--------------
+
+By default, every ``MongoCollection`` has ``Codec``s registered for
+three classes:
+
+- ``Document``
+- ``BasicDBObject``
+- ``BsonDocument``
+
+Custom Codecs
+-------------
+
+Applications are free to register ``Codec`` implementations for other
+classes by adding them to the ``CodecRegistry``. You can configure
+``CodecRegistry`` instances at the ``MongoClient``, ``MongoDatabase``,
+and ``MongoCollection`` levels.
+
+To register a ``Codec`` for a ``MongoClient``, use the
+:java-api-sync:`MongoClientSettings
+<com/mongodb/MongoClientSettings.html>` class:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoClientSettings settings = MongoClientSettings.builder()
+        .codecRegistry(codecRegistry).build();
+   MongoClient client = MongoClients.create(settings);
+
+To register a ``Codec`` for a ``MongoDatabase``, use the
+:java-sync-api:`withCodecRegistry()
+<com/mongodb/client/MongoDatabase.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoDatabase database = client.getDatabase("exampleDatabase")
+                               .withCodecRegistry(codecRegistry);
+
+To register a ``Codec`` for a ``MongoCollection``, use the
+:java-sync-api:`withCodecRegistry()
+<com/mongodb/client/MongoCollection.html#withCodecRegistry(org.bson.codecs.configuration.CodecRegistry)>`
+method:
+
+.. code-block:: java
+
+   CodecRegistry codecRegistry =
+      CodecRegistries.fromRegistries(MongoClientSettings.getDefaultCodecRegistry(),
+                fromProviders(PojoCodecProvider.builder().automatic(true).build()));
+   MongoCollection<Document> collection = database.getCollection("exampleCollection")
+                                               .withCodecRegistry(codecRegistry);
+
+UUID Class
+----------
+
+By default, the driver encodes instances of `UUID
+<https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/util/UUID.html>`__
+using a byte ordering that is not compatible with other MongoDB drivers.
```

Comment:
I would probably separate the compatibility info into another sentence just to create a logical separation between statements.

Reply:
I'm not sure if there's any way to separate these logically -- the whole point of the statement is that "the default encoding of uuid in the java driver is not compatible with any other drivers". Without the compatibility piece, the statement just tells readers that the java driver encodes uuids, which doesn't give any new information to users who know that UUID is a Bson type that the driver has to support.